controllers/rbac/directoryrolebinding/integration/integration_test.go

```diff
@@ -58,73 +55,42 @@ func newUser(name string) rbacv1.Subject {
 
 var _ = Describe("Reconciler", func() {
 	var (
-		ctx       context.Context
-		cancel    func()
-		namespace string
-		labels    map[string]string
-		teardown  func()
-		mgr       manager.Manager
-		calls     chan integration.ReconcileCall
-		groups    map[string][]string
+		namespaceName string
+		labels        map[string]string
 	)
 
 	BeforeEach(func() {
-		ctx, cancel = context.WithTimeout(context.Background(), time.Minute)
-		namespace, teardown = integration.CreateNamespace(clientset)
+		namespaceName = uuid.New().String()
+
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		}
+
 		labels = map[string]string{
 			"repo":        "foo-repo",
 			"environment": "foo-env",
 		}
-		mgr = integration.StartTestManager(ctx, cfg)
-		groups = map[string][]string{}
-
-		// Create a fake provider that recognises GoogleGroup kinds for testing purposes
-		provider := directoryrolebinding.DirectoryProvider{}
-		provider.Register(rbacv1alpha1.GoogleGroupKind, directoryrolebinding.NewFakeDirectory(groups))
-
-		integration.MustController(
-			directoryrolebinding.Add(
-				ctx,
-				kitlog.NewLogfmtLogger(GinkgoWriter),
-				mgr,
-				provider,
-				time.Duration(0), // don't test our caching/re-enqueue here
-				func(opt *controller.Options) {
-					opt.Reconciler, calls = integration.CaptureReconcile(
-						opt.Reconciler,
-					)
-				},
-			),
+
+		By("Creating test namespace: " + namespaceName)
+		Expect(mgr.GetClient().Create(context.TODO(), namespace)).NotTo(
+			HaveOccurred(), "failed to create test namespace",
 		)
 
 		By("Creating fixture 'admin' role")
-		Expect(mgr.GetClient().Create(ctx, newAdminRole(namespace))).NotTo(
+		Expect(mgr.GetClient().Create(context.TODO(), newAdminRole(namespaceName))).NotTo(
 			HaveOccurred(), "failed to create test 'admin' Role",
 		)
 	})
 
-	AfterEach(func() {
-		cancel()
-		teardown()
-	})
-
 	Context("With all@ and platform@", func() {
-		BeforeEach(func() {
```

Comment:
Not entirely sure I'm following this, have some of the tests been moved to `suite_test.go`?

Reply:
Never mind, I think I follow now.

Reply:
I can't help but feel that perhaps moving out some of the larger changes, e.g the logging rework, testing changes etc to separate PRs would make this easier to review & follow..